### PR TITLE
Issue #51 Fix .jar creation

### LINKED ISSUE #51: Create a release process
Automatically generate a JAR file when creating a release with a tag

### PR DESCRIPTION
**Pull request by superjeffcplusplus**
I adapted the pom.xml so that we can build a `.jar` file containing the resources of the application and its dependencies.
I had to modify the way we open the resources file to be compatible with the run of the application from `.jar` file.
Close #51

### CODE REVIEW

File: src/test/resources/testFiles/test-config/page-config-valid.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "title" : "title example",
+  "author" : "Eliott Chytil",
+  "datte" : "21-03-10"
```
Comment (Luhrel) [inline src/test/resources/testFiles/test-config/page-config-valid.json:4]: `datte`?

File: src/test/resources/testFiles/test-config/page-config-missing-date.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "title" : "title example",
+  "author" : "Eliott Chytil",
+  "date" : "21-03-10"
```
Comment (Luhrel) [inline src/test/resources/testFiles/test-config/page-config-missing-date.json:4]: This file seems valid

File: src/test/java/ch/heigvd/dil/utils/parsers/ConfigGeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    String schemaPath = "schema/site-config-schema.json";
+    try {
+      ConfigGenerator<Site.Config> generator =
+          new ConfigGenerator<>(validConfig.toString(), schemaPath, Site.Config.class);
+    } catch (ValidationException e) {
+      thrown = true;
+    }
+    assertFalse(thrown);
+  }
+}
```
Comment (Luhrel) [inline src/test/java/ch/heigvd/dil/utils/parsers/ConfigGeneratorTest.java:33]: Please create a test for every content of `testFiles/test-config/*` -> move content into a String.

**Review by Luhrel — CHANGES_REQUESTED** [review]

File: src/test/java/ch/heigvd/dil/utils/parsers/ConfigGeneratorTest.java (hunk 3 above)
Comment (superjeffcplusplus) [reply src/test/java/ch/heigvd/dil/utils/parsers/ConfigGeneratorTest.java:50]: done

File: src/test/resources/testFiles/test-config/page-config-missing-date.json (hunk 2 above)
Comment (superjeffcplusplus) [reply src/test/resources/testFiles/test-config/page-config-missing-date.json]: deleted

File: src/test/resources/testFiles/test-config/page-config-valid.json (hunk 1 above)
Comment (superjeffcplusplus) [reply src/test/resources/testFiles/test-config/page-config-valid.json]: file deleted

**Review by Luhrel — APPROVED** [review]

**Review by Webblitchy — APPROVED** [review]

**Pull request merged**